### ISSUE TITLE
Task 1: Train and save the RandomForestRegressor

### ISSUE THREAD

**Issue by SergeATX** [TriangulumAustrale/Scikit-learn#3]
## Summary
Write `backend/train.py`: load the California Housing dataset, train a `RandomForestRegressor`, evaluate it, and save the model + feature importances as artifacts the API will load.

Full spec: see the Data & Model section (§6) of the [Project Plan](https://github.com/TriangulumAustrale/Scikit-learn/issues/2).

## Instructions
```python
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import r2_score, mean_absolute_error
import joblib, json

data = fetch_california_housing(as_frame=True)
X, y = data.data, data.target

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = RandomForestRegressor(n_estimators=100, random_state=42)
model.fit(X_train, y_train)

preds = model.predict(X_test)
print("R2:", r2_score(y_test, preds))
print("MAE:", mean_absolute_error(y_test, preds))

joblib.dump(model, "backend/model/model.joblib")

importances = dict(zip(X.columns, model.feature_importances_.tolist()))
with open("backend/model/feature_importances.json", "w") as f:
    json.dump(importances, f, indent=2)
```

## Acceptance Criteria
- [ ] `backend/train.py` runs end-to-end with `python train.py`
- [ ] Prints R² and MAE on the held-out test set
- [ ] Produces `backend/model/model.joblib`
- [ ] Produces `backend/model/feature_importances.json`
- [ ] `backend/model/` added to `.gitignore` is NOT needed — artifacts should be committed (small enough to bundle for Lambda deploy later)

## Out of scope
Hyperparameter search, model comparison, cross-validation — single model, single training run per the plan's scope constraints (§3).
